src/dbl_linked_list.py

```
@@ -0,0 +1,150 @@
+                for value in maybe_an_iterable:
+                    self.push(value)
+            except TypeError:
+                self.push(maybe_an_iterable)
+
+    def push(self, value):
+        """Add a node at the beginning of list and reassign head."""
+        new = DblNode(value)
+        if self._size > 0:
+            old = self.head
+            self.head = new
+            new.nxt = old
+            old.prev = new
+        else:
+            self.head = new
+            self.tail = self.head
+        self._size += 1
+
+    def pop(self):
+        """Remove head node from DLL and return its value."""
+        if self.head is None:
+            raise IndexError("Cannot pop from an empty Double Linked List.")
+        val = self.head.value
+        if self._size == 1:
+            self.head = None
+            self.tail = None
+        else:
+            self.head = self.head.nxt
+            self.head.prev = None
+        self._size -= 1
+        return val
+
+    def shift(self):
+        """Remove tail node from DLL and return its value."""
+        if self.tail is None:
+            raise IndexError("Cannot shift from an empty Double Linked List.")
+        val = self.tail.value
+        if self._size == 1:
+            self.head = None
+            self.tail = None
+        else:
+            self.tail = self.tail.prev
+            self.tail.nxt = None
+        self._size -= 1
+        return val
+
+    def append(self, value):
+        """Append a new node at the end of list and reassign tail."""
+        new = DblNode(value)
+        if self._size > 0:
+            old = self.tail
+            self.tail = new
+            new.prev = old
+            old.nxt = new
+        else:
+            self.tail = new
+            self.head = self.tail
+        self._size += 1
+
+    def remove(self, value):
+        """Remove the given node from the DLL."""
+        r_node = [node for node in self._iterate_from(self.head) if node.value == value]
```

No need to iterate and construct a list here, since you're iterating through the whole data structure once later in the function. If the iteration goes all the way through without returning anything, then you can raise an error.